File: backend/src/main/java/com/erp/service/NotificationService.java
package com.erp.service;

import com.erp.dto.response.NotificationResponseDto;
import com.erp.dto.response.PaginatedResponse;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    PaginatedResponse<NotificationResponseDto> getMyNotifications(Pageable pageable);
    long getUnreadCount();
    void markAsRead(Long id);
    void markAllAsRead();
}

File: frontend/src/services/notificationService.js
import api from './api'

export const notificationService = {
  getMyNotifications: async (page = 0, size = 15) => {
    const response = await api.get('/api/notifications', { params: { page, size } })
    return response.data
  },

  getUnreadCount: async () => {
    const response = await api.get('/api/notifications/unread-count')
    return response.data.unreadCount
  },

  markAsRead: async (id) => {
    const response = await api.put(`/api/notifications/${id}/read`)
    return response.data
  },

  markAllAsRead: async () => {
    const response = await api.put('/api/notifications/read-all')
    return response.data
  }
}

export default notificationService
